metpy/calc/indices.py

```diff
@@ -51,3 +51,49 @@ def precipitable_water(dewpt, p, top=400 * units('hPa')):
     pw = -1. * (np.trapz(w.magnitude, pres_layer.magnitude) * (w.units * pres_layer.units) /
                 (g * rho_l))
     return pw.to('millimeters')
+
+
+@exporter.export
+@check_units('[speed]', '[speed]', '[pressure]', '[length]')
+def mean_wind_pressure_weighted(u, v, p, heights=None, bottom=None, depth=None):
+    r"""Calculate pressure-weighted mean wind through a layer.
+
+    Layer top and bottom specified in height or pressure.
+
+    Parameters
+    ----------
+    u : `pint.Quantity`
+        U-component of wind.
+    v : `pint.Quantity`
+        V-component of wind.
+    p : `pint.Quantity`
+        Atmospheric pressure profile
+    heights : `pint.Quantity`, optional
+        Heights from sounding
+    bottom: `pint.Quantity`, optional
+        The bottom of the layer in either the provided height coordinate
+        or in pressure. Don't provide in meters AGL unless the provided
+        height coordinate is meters AGL. Default is the first observation,
+        assumed to be the surface.
+    depth: `pint.Quantity`, optional
+        The depth of the layer in meters or hPa.
+
+    Returns
+    -------
+    `pint.Quantity`
+        u_mean: u-component of layer mean wind, in m/s
+    `pint.Quantity`
+        v_mean: v-component of layer mean wind, in m/s
+
+    """
+    u = u.to('meters/second')
+    v = v.to('meters/second')
+    layer_p, layer_u, layer_v = get_layer(p, u, v, heights=heights,
+                                          bottom=bottom, depth=depth)
+
+    u_mean = (np.trapz(layer_u * layer_p, x=layer_p) /
```

Comment:
I don't think a helper is necessary--but we shouldn't be calculating the denominator for the mean twice since it's the same for both. Instead calculate the integral of p once and use in both.

Also....isn't `np.trapz(layer_p, x=layer_p)` just `0.5 * (layer_p[-1]**2 - layer_p[0]**2)`? (When you use the latter, be sure to add a comment indicating that this is just the integral of pressure.)